source/features/select-all-notifications-shortcut.tsx

```diff
@@ -1,22 +1,34 @@
-import elementReady from 'element-ready';
+import select from 'select-dom';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
-import onElementRemoval from '../helpers/on-element-removal';
 
-async function init(signal: AbortSignal): Promise<void> {
-	const selectAllNotifications = await elementReady('.js-notifications-mark-all-prompt');
-	if (selectAllNotifications) { // Notifications page may be empty
-		selectAllNotifications.dataset.hotkey = 'a';
-		await onElementRemoval(selectAllNotifications, signal); // "Select all" checkbox will be replaced if there's more notifications to load #4199
-		void init(signal);
+function runShortcut(event: KeyboardEvent): void {
+	if (
+		event.key === 'a'
+		&& !event.ctrlKey
+		&& !event.metaKey
+		&& !event.isComposing
+	) {
+		select('.js-notifications-mark-all-prompt')!.click();
 	}
 }
 
+function init(signal: AbortSignal): void {
+	// Listen to the shortcut ourselves instead of attaching it to the checkbox #5569
+	document.body.addEventListener('keypress', runShortcut, {signal});
```

Comment:
We could potentially also have a helper that registers a callback with the native `data-hotkey`, e.g.


```js
function registerHotkey(hotkey, handler) {
	document.body.append(<button data-hotkey={hotkey} onClick={handler}>)
	return () => select(`[data-hotkey="${hotkey}"`)?.remove()
}
```